### ISSUE TITLE
Frequent crashes during the use of Oracle DB

### ISSUE THREAD

**Issue by VadimMuhtarov** [TablePlus/TablePlus#3300]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
Oracle 19

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
6.0.3 (555), Sonoma 14.5 (23F79)

3. The steps to reproduce this issue:
No exact steps, just use Oracle DB and it will crash after some time

[TablePlus-2024-05-16-091347.log](https://github.com/TablePlus/TablePlus/files/15331193/TablePlus-2024-05-16-091347.log)
[TablePlus-2024-05-10-125536.log](https://github.com/TablePlus/TablePlus/files/15331194/TablePlus-2024-05-10-125536.log)
[TablePlus-2024-05-13-123500.log](https://github.com/TablePlus/TablePlus/files/15331195/TablePlus-2024-05-13-123500.log)
[TablePlus-2024-05-14-094354.log](https://github.com/TablePlus/TablePlus/files/15331196/TablePlus-2024-05-14-094354.log)

**Comment by huyphams** [comment]
Hi @VadimMuhtarov thank you. We're using the Oracle Instant Client as the native driver, which is quite bad. It is only support intel base x86_64 which means TablePlus beta you're using running on Rosetta 2.

We will move away from Oracle Instant Client in the future updates with a better, lighter driver and support Apple Silicon Chip natively.

**Comment by VadimMuhtarov** [comment]
Do you mean TablePlus updates or new Bolt SQL client?

**Comment by huyphams** [comment]
TablePlus and Bolt SQL are two apps with differences feature set. This update has nothing todo with Bolt SQL.

**Comment by huyphams** [comment]
Hi, please try this patch and see how it goes, TablePlus now support Oracle thin with ARM native driver, no need Rosetta 2.
https://www.dropbox.com/scl/fi/rnzl329j9llknuoj0c0h1/TablePlus.dmg?rlkey=fykl1eshbl1ylnn48s5ic7rau&dl=0

**Comment by VadimMuhtarov** [comment]
Can't connect now, same database, same credentials. Should I create new bug ticket?
<img width="267" alt="Screenshot 2024-06-21 at 07 46 33" src="https://github.com/TablePlus/TablePlus/assets/17072773/65b03159-6ef0-4288-8b40-b20274b302d9">

**Comment by huyphams** [comment]
Which Oracle version are you using @VadimMuhtarov ?

**Comment by VadimMuhtarov** [comment]
Oracle 19

**Comment by huyphams** [comment]
Let me add the error message and see what was wrong.

**Comment by huyphams** [comment]
You can download again using the same link above @VadimMuhtarov

**Comment by huyphams** [comment]
ah my bad, building a new build.

**Comment by huyphams** [comment]
You can try again, the same link above

**Comment by VadimMuhtarov** [comment]
Connected, will be testing

**Comment by huyphams** [comment]
Crashes are expected but it is easy to fix them now, please report if you have found any. Thanks

**Comment by VadimMuhtarov** [comment]
Crashed in background again
[TablePlus-2024-06-21-101334.logs.log](https://github.com/user-attachments/files/15924119/TablePlus-2024-06-21-101334.logs.log)

**Comment by huyphams** [comment]
Hi @VadimMuhtarov just updated the build you can download again using the same link. It also would be great if you can attach the report from Apple report instead of the json. it's hard to see the crash log.

**Comment by VadimMuhtarov** [comment]
Just got another 2 crashes on the latest build. Don't know what you mean under "Apple report", but these are from Crash Reports in Console utility
[TablePlus-2024-06-22-082048.ips.zip](https://github.com/user-attachments/files/15935408/TablePlus-2024-06-22-082048.ips.zip)
[TablePlus-2024-06-22-082418.ips.zip](https://github.com/user-attachments/files/15935409/TablePlus-2024-06-22-082418.ips.zip)

**Comment by huyphams** [comment]
@VadimMuhtarov thanks, after the app crash, Apple usually shows a report popup, you can click detail to see the crash call stack.

**Comment by huyphams** [comment]
btw, I think your two recent crash logs are good enough, thank you very much!

**Comment by VadimMuhtarov** [comment]
Just copied a fresh log from that popup if needed 
[crash.log](https://github.com/user-attachments/files/15937200/crash.log)

**Comment by VadimMuhtarov** [comment]
Most of the crashes are random. The last one was after I ran 

SELECT
	*
FROM
	table_1 A
	JOIN table_2 B ON A.ID = B.ID
	JOIN table_3 C ON B.ID = C.ID;

**Comment by huyphams** [comment]
Just updated the new build with more debugging symbol and support Oracle XML TYPE decoding. If you have a chance, please try it. (same link above).

**Comment by VadimMuhtarov** [comment]
[TablePlus-2024-06-22-220936.ips.zip](https://github.com/user-attachments/files/15938063/TablePlus-2024-06-22-220936.ips.zip)

**Comment by huyphams** [comment]
Just updated the build which prevent TablePlus from the crash on your report.

**Comment by VadimMuhtarov** [comment]
This crash has been fixed, thanks, I'll keep looking.

**Comment by huyphams** [comment]
thanks for confirming, going to publish new official beta build now.

**Comment by samuelcseto** [comment]
Hi @huyphams, I am getting crashes on the most recent beta version. (Build 569) I have narrowed it down to I guess too large return response from server. If I get all 33k rows of data from table with just 1 column retrieved, it's okay. If I get all columns, I can only get around 100 rows. If I don't limit it using ROWNUM, the app crashes. I got the error report dialog a few times, but now the app just closes. Any ideas what to do?
